atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -339,6 +356,32 @@ private void registerPoolMetric(CassandraClientPoolHostLevelMetric metric, Gauge
         poolMetrics.registerPoolMetric(metric, gauge, poolNumber);
     }
 
+    private static final class TimingOutEvictionPolicy<T> implements EvictionPolicy<T> {
+        private final EvictionPolicy<T> delegate;
+        private final Clock clock;
+        private volatile Instant lastEviction;
```

Comment:
Can we use `AtomicReference` and have the update operation take the latest one? Different threads writing their values to `lastEviction` don't actually read it, and so unless there is some other synchronisation elsewhere I think this might actually go backwards